libqfieldsync/project.py

```diff
@@ -158,6 +159,29 @@ def geofencing_should_prevent_digitizing(self, value):
             "qfieldsync", ProjectProperties.GEOFENCING_SHOULD_PREVENT_DIGITIZING, value
         )
 
+    @property
+    def map_themes_active_layer(self):
+        entries, _ = self.project.readListEntry(
+            "qfieldsync", ProjectProperties.MAP_THEMES_ACTIVE_LAYER
+        )
+
+        map_themes_active_layer = {}
+        for entry in entries:
+            details = entry.split("}|~~|{")
+            if len(details) == 2:
+                map_themes_active_layer[details[0]] = details[1]
+
+        return map_themes_active_layer
+
+    @map_themes_active_layer.setter
+    def map_themes_active_layer(self, value):
+        entries = []
+        for key in value:
+            entries.append(f"{key}}}|~~|{{{value[key]}")
+        self.project.writeEntry(
+            "qfieldsync", ProjectProperties.MAP_THEMES_ACTIVE_LAYER, entries
+        )
+
```

Comment:
```suggestion
    @property
    def map_themes_active_layer(self):
        entries_json, _ = self.project.readListEntry(
            "qfieldsync", ProjectProperties.MAP_THEMES_ACTIVE_LAYER
        )

	try:
	    entries = json.reads(entries_json)
	except Exception:
	    entries = {}

        return entries

    @map_themes_active_layer.setter
    def map_themes_active_layer(self, value):
        self.project.writeEntry(
            "qfieldsync", ProjectProperties.MAP_THEMES_ACTIVE_LAYER,
             json.dumps(value),
        )
```

I think this is much more readable and maintainable, unless I miss something from the original implementation?

Reply:
@suricactus , that works, I'll update.